apps/emqx_authn/src/enhanced_authn/emqx_enhanced_authn_scram_mnesia.erl

```diff
@@ -169,84 +177,80 @@ authenticate(
 authenticate(_Credential, _State) ->
     ignore.
 
-destroy(#{user_group := UserGroup}) ->
+destroy(State) ->
+    trans(fun ?MODULE:do_destroy/1, [State]).
+
+do_destroy(#{user_group := UserGroup}) ->
```

Comment:
Just thinking out loud (don't think there's anything to be done in this PR): this suggests we might eventually need to do something like the `bpapi` for transactions, indeed?  

Because if `State` is changed between releases (say, the `user_group` key no longer exists), then the core that will execute this transaction has to be compatible with the new type.

Reply:
for this specific case, a safer option is to move this pattern match outside of the transaction function.
the transaction fun's args should be as primitive (close to the actual table key) as possible.

Reply:
I tried to make a quick prototype with BPAPI and it was too ugly to proceed. 
There is a conceptual problem: RPC is meant to ask the remote node to do something, hence versioning and whatnot. 
In this case we simply want the remote node to run whatever function we supply. The core node shouldn't really care what function it runs.